### PR TITLE
 Fix flaky integration test

### LINKED ISSUE #3352: Should_successfully_reconcile_AWCluster_creation_with_unmanaged_VPC test is flaky
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Created 2 PRs. Observed twice this test failing randomly: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/3351/pull-cluster-api-provider-aws-test/1507369944281518080

**What did you expect to happen:**

All tests to pass.

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/area testing
/kind flake

**What this PR does / why we need it**:
This PR fixes flaky integration test in AWSCluster controller. The sorting of tags by keys are in place in the test, this PR is just making it a bit more generic.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3352 

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [X] adds unit tests
- [ ] adds or updates e2e tests

**Release Notes**:
```release-note
None
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
The tests looks good now.
cc @sedefsavas 
P.S the failures we are seeing are because of below error and not related to integration tests.
E0328 11:55:52.658126   22531 logr.go:279] controller-runtime/test-env "msg"="unable to start the controlplane" "error"="timeout waiting for process kube-apiserver to start"  "tries"=0
E0328 11:55:54.692573   22531 logr.go:279] controller-runtime/test-env "msg"="unable to start the controlplane" "error"="timeout waiting for process kube-apiserver to start successfully (it may have failed to start, or stopped unexpectedly before becoming ready)"  "tries"=1
E0328 11:55:56.588849   22531 logr.go:279] controller-runtime/test-env "msg"="unable to start the controlplane" "error"="timeout waiting for process kube-apiserver to start successfully (it may have failed to start, or stopped unexpectedly before becoming ready)"  "tries"=2
E0328 11:55:58.628690   22531 logr.go:279] controller-runtime/test-env "msg"="unable to start the controlplane" "error"="timeout waiting for process kube-apiserver to start successfully (it may have failed to start, or stopped unexpectedly before becoming ready)"  "tries"=3
E0328 11:56:00.296059   22531 logr.go:279] controller-runtime/test-env "msg"="unable to start the controlplane" "error"="timeout waiting for process kube-apiserver to start successfully (it may have failed to start, or stopped unexpectedly before becoming ready)"  "tries"=4
panic: unable to start control plane itself: failed to start the controlplane. retried 5 times: timeout waiting for process kube-apiserver to start successfully (it may have failed to start, or stopped unexpectedly before becoming ready)

>

File: controllers/helpers_test.go [hunk 1]
```diff
@@ -162,23 +162,19 @@ func mockedCreateLBCalls(t *testing.T, m *mock_elbiface.MockELBAPIMockRecorder)
...
+	m.AddTags(gomock.AssignableToTypeOf(&elb.AddTagsInput{})).Return(&elb.AddTagsOutput{}, nil).Do(
+		func(actual *elb.AddTagsInput) {
+			sortTagsByKey := func(tags []*elb.Tag) {
+				sort.Slice(tags, func(i, j int) bool {
+					return *(tags[i].Key) < *(tags[j].Key)
+				})
+			}
 
-		sortTagsByKey(actual.Tags)
-		sortTagsByKey(expectedTags)
```
Comment (sedefsavas) [inline controllers/helpers_test.go:176]: Why don't we sort the expected keys?

What is the fix in this PR?

Comment (Ankitasw) [reply controllers/helpers_test.go:176]: I have sorted it manually in the declaration itself, only change is we are expecting AddTagsInput only with type check now and not with the actual input fields.

Comment (Ankitasw) [reply controllers/helpers_test.go:176]: This has been followed in other files as well, so followed the same, and those tests doesn't flake.

Comment (Ankitasw) [reply controllers/helpers_test.go:176]: The sorttags func works fine, i think the main catch here is the expected mock input for AddTagsInput. It used to first compare the expectedTags and then do sorting, that's why it used to fail sometimes, now we dont rely on the input parameters itself and just check AssignableOfType AddTagsInput.

Comment (Ankitasw) [reply controllers/helpers_test.go:176]: So even if we leave th sortTags(expectedTags) as is, it will work.

**Pull request merged**